playwright/pages/ChangesPage.ts

```diff
@@ -20,23 +25,57 @@ export class ChangesPage extends SliceMachinePage {
      * Static locators
      */
     this.breadcrumbLabel = this.breadcrumb.getByText("Changes");
+    this.loginButton = this.body.getByText("Log in to Prismic");
+    this.pushChangesButton = page.getByText("Push Changes");
+    this.pushedMessaged = page.getByText(
+      "All slices and types have been pushed",
+    );
+    this.notLoggedInTitle = page.getByText("It seems like you are logged out");
+    this.notAuthorizedTitle = page.getByText(
+      "It seems like you don't have access to this repository",
+    );
+    this.blankSlateTitle = page.getByText("Everything up-to-date");
```

Comment:
If we use these queries elsewhere, should we abstract them to a set of helpers? (If we don't use it elsewhere, then we can skip the abstraction, of course).

Reply:
Should these be evaluated immediately? I'm wondering if we'll run into a race condition in tests, such as this:

```ts
  async goto() {
    await this.page.goto("/changes");
    await expect(this.breadcrumbLabel).toBeVisible();
    // What happens if breadcrumbLabel didn't exist until we loaded `/changes`?
  }
```

I would expect something like this instead:

```ts
  async goto() {
    await this.page.goto("/changes");
    await expect(this.getBreadcrumbLabel()).toBeVisible();
  }
```

Reply:
> If we use these queries elsewhere

Hum, not sure to understand. It's already an abstraction here with POM model. ChangesPage is the abstraction here. these locators are specific to ChangesPage. `    this.blankSlateTitle = page.getByText("Everything up-to-date");
` is only for here.

Reply:
> Should these be evaluated immediately? I'm wondering if we'll run into a race condition in tests, such as this:
Oh no need with Playwright to do this. 

`await expect(this.breadcrumbLabel).toBeVisible();` is only checked at the moment the line is processed. It's how you're writing test in Playwright, no need to wrap everything in function. Locator are already function in a sense. 

see: https://playwright.dev/docs/locators

let me know If I didn't really answer your question